.github/workflows/main.yml

```diff
@@ -118,11 +118,23 @@ jobs:
         run: ./ci/docker-build.sh ${{ matrix.artifact }}
         if: runner.os == 'Linux'
 
-      # Use a shorter build directory than the default on Windows to avoid
-      # hitting path length and command line length limits. See
-      # WebAssembly/wasi-libc#514
+      # Setup the VS Developoer Prompt environment variables to explicitly use
+      # MSVC to compile LLVM as that avoids extra runtime dependencies
+      # msys/mingw might bring.
+      #
+      # As of 2024-07-22 this sha is the "v1" tag.
+      - uses: ilammy/msvc-dev-cmd@0b201ec74fa43914dc39ae48a89fd1d8cb592756
```

Comment:
I was debating between the need for this action versus the risk: it looks like ~17k other projects depend on this, it has more than a few contributors, and the action is relatively simple &mdash; if it solves the problem then let's go for it.

Reply:
Yeah I was wrestling with the same. I spend 30 min or so googling around to see if there were other common solutions to this but everything pointed to a github action one way or another so I opted to do that with a specific sha to at least be insulated against changes.

Reply:
personally, I would fork any dep like this (in case the repo gets deleted, repurposed, ...)